### PR TITLE
release: dev → main (0.26.13 – debug logs, backlog refine, OPSX, fixes)

### LINKED ISSUE #158: [Change] Add debug logs under ~/.specfact/logs with rich operation metadata
## Why

When `--debug` is enabled globally, users and developers need consistent debug logs to diagnose IO, file handling, and API issues. Today debug output is only printed to the console via `debug_print()` and is not persisted; there is no user-level log directory and no structured operation metadata (status, return, error) for file or API operations. Storing debug logs under `~/.specfact/logs` and including rich operation metadata makes it possible to identify issues when something does not work as expected without cluttering normal CLI output.

## What Changes

- **NEW**: User-level debug log directory `~/.specfact/logs` (create on first use when debug is enabled; mode 0o755). Add `get_specfact_home_logs_dir()` in `logger_setup.py` returning this path.
- **EXTEND**: When `--debug` is set, route `debug_print()` output to both console (current) and a debug log file under `~/.specfact/logs` (e.g. rotating `specfact-debug.log`).
- **NEW**: Optional helper `debug_log_operation(operation, target, status, error=None, extra=None)` that no-ops when debug is off and when debug is on writes structured metadata to the debug log file and optionally calls `debug_print()`.
- **EXTEND**: In key places (file IO, API calls), when debug is on, log operation metadata: operation type, target (path/URL redacted), status/result, error snippet; use `LoggerSetup.redact_secrets` for any response/request bodies.
- **EXTEND**: Adapters (ADO, GitHub) and commands (backlog refine, init) log file/API operation metadata when debug is enabled.
- **NOTE**: `get_runtime_logs_dir()` remains unchanged for existing callers; `~/.specfact/logs` is used only for the dedicated debug session log.

## Acceptance Criteria

- [ ] With `--debug`, a debug log file exists under `~/.specfact/logs` and receives all `debug_print()` output plus structured operation metadata for file and API operations.
- [ ] Without `--debug`, no `~/.specfact/logs` directory is created and behavior is unchanged.
- [ ] Secrets (tokens, keys) are redacted in debug logs (reuse `LoggerSetup.redact_secrets`).
- [ ] CLI help for `--debug` mentions log location and purpose; CHANGELOG updated.

---
*OpenSpec Change Proposal: `add-debug-logs-specfact-home`*

### PR DESCRIPTION
**Pull request by djm81**
# Description

Release PR: **dev → main** to promote changes from the development branch to production. This brings all dev-only changes into main for release (PyPI/container).

**Summary of changes (since last main):**

- **0.26.13** – Debug log parity for `specfact upgrade`: "up to date" success path now writes to `~/.specfact/logs/specfact-debug.log` when `--debug` is set.
- **0.26.12** – Debug logs under `~/.specfact/logs`: `debug_print()` and `debug_log_operation()`; log file at `~/.specfact/logs/specfact-debug.log`; adapters and commands log operation metadata when `--debug`; docs and tests. Closes #158.
- **0.26.11** – Backlog refine `--import-from-tmp`: parser and import flow; unit tests.
- **0.26.10** – OpenSpec OPSX migration docs; config.yaml (OPSX) vs project.md (legacy); version bump fix.
- **0.26.9** – GitHub remote detection (`ssh://`, `git://`); code scanning mitigations (ReDoS, URL sanitization, workflow permissions).
- **0.26.8** – ADO field mappings (acceptance criteria, assignee); `backlog map-fields`; compare SSH hostnames case-insensitively.
- **Earlier** – Startup performance (metadata tracking, update command); OpenSpec/workflow commands; contribution standards.

**Fixes** #158 (via #159), #148, #145, #142, #156

**New Features** #158 (debug logging)

**Contract References**: No new `@icontract` decorators in this release; existing contracts unchanged. Debug helpers are no-ops when debug is off.

## Type of Change

Please check all that apply:

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🔒 Contract enforcement (adding/updating `@icontract` decorators)
- [x] 🧪 Test enhancement (scenario tests, property-based tests)
- [ ] 🔧 Refactoring (code improvement without functionality change)

### Contract Validation

- [ ] **Runtime contracts added/updated** (`@icontract` decorators on public APIs)
- [ ] **Type checking enforced** (`@beartype` decorators applied)
- [ ] **CrossHair exploration** completed: `hatch run contract-test-exploration`
- [x] **Contract violations** reviewed and addressed

### Test Execution

- [x] **Contract validation**: `hatch run contract-test-contracts` ✅
- [ ] **Contract exploration**: `hatch run contract-test-exploration` ✅
- [x] **Scenario tests**: `hatch run contract-test-scenarios` ✅
- [x] **Full test suite**: `hatch run contract-test-full` ✅

### Test Quality

- [x] **CLI commands tested** with typer test client
- [ ] **Edge cases covered** with Hypothesis property tests
- [x] **Error handling tested** with invalid inputs
- [x] **Rich console output verified** manually or with snapshots

## How Has This Been Tested?

All changes were tested and reviewed on dev via PRs (#159, #156, #148, #145, #142). Contract validation and scenario tests pass. Debug logging verified manually (`specfact --debug sdd list`, `specfact --debug upgrade --check-only`) and log file inspected at `~/.specfact/logs/specfact-debug.log`.

### Manual Testing

- [x] Tested CLI commands manually
- [x] Verified rich console output
- [x] Tested with different input scenarios
- [x] Checked error messages for clarity

### Automated Testing

- [x] Contract validation passes
- [x] Scenario tests cover user workflows
- [x] All existing tests still pass

### Test Environment

- Python version: 3.12
- OS: Linux (Ubuntu)

## Checklist

- [x] My code follows the style guidelines (PEP 8, ruff format, isort)
- [x] I have performed a self-review of my code
- [x] I have added/updated contracts (`@icontract`, `@beartype`) — N/A for this release PR
- [x] I have added/updated docstrings (Google style)
- [x] I have made corresponding changes to documentation
- [x] My changes generate no new warnings (basedpyright, ruff, pylint)
- [x] All tests pass locally
- [x] I have added tests that prove my fix/feature works
- [x] Any dependent changes have been merged (all merged to dev)

## Quality Gates Status

- [x] **Type checking** ✅ (`hatch run type-check`)
- [x] **Linting** ✅ (`hatch run lint`)
- [x] **Contract validation** ✅ (`hatch run contract-test-contracts`)
- [ ] **Contract exploration** ✅ (`hatch run contract-test-exploration`)
- [x] **Scenario tests** ✅ (`hatch run contract-test-scenarios`)

## Screenshots/Recordings (if applicable)

N/A for release PR. Debug log output was verified in #159.

### CODE REVIEW

File: src/specfact_cli/runtime.py [hunk 1]
```diff
@@ -175,18 +191,126 @@ def get_configured_console() -> Console:
...
+        )
+        _debug_file_handler.setLevel(logging.DEBUG)
+        _debug_file_handler.setFormatter(logging.Formatter(DEBUG_LOG_FORMAT, datefmt=DEBUG_LOG_DATEFMT))
+        _debug_logger = logging.getLogger("specfact.debug")
+        _debug_logger.setLevel(logging.DEBUG)
+        _debug_logger.propagate = False
+        _debug_logger.handlers.clear()
+        _debug_logger.addHandler(_debug_file_handler)
+    except (OSError, PermissionError):
+        _debug_file_handler = None
```
[the remark this answers is not in the document]
Comment (djm81) [reply src/specfact_cli/runtime.py]: Will be checked and fixed
